eo-runtime/src/main/java/org/eolang/PhDefault.java

```diff
@@ -117,46 +111,53 @@ public int hashCode() {
 
     @Override
     public String φTerm() {
-        if (this.terms.get() == null) {
-            this.terms.set(new HashSet<>());
+        if (PhDefault.TERMS.get() == null) {
+            PhDefault.TERMS.set(new HashSet<>());
         }
-        if (this.terms.get().contains(this.vertex)) {
-            return String.format("ν%d", this.vertex);
-        }
-        this.terms.get().add(this.vertex);
-        final List<String> list = new ArrayList<>(this.attrs.size());
-        for (final Map.Entry<String, Attr> ent : this.attrs.entrySet().stream().filter(
-            e -> Arrays.asList("σ", "ρ").contains(e.getKey())
-        ).collect(Collectors.toList())) {
-            final String attr = String.format(
-                "%s ↦ %s",
-                ent.getKey(),
-                ent.getValue().φTerm()
-            );
-            list.add(attr);
-        }
-        this.terms.get().remove(this.vertex);
-        Collections.sort(list);
-        String txt = this.getClass().getSimpleName();
-        final XmirObject xmir = this.getClass().getAnnotation(XmirObject.class);
-        if (xmir != null) {
-            txt = xmir.oname();
-            if ("@".equals(txt)) {
-                txt = "φ";
+        String txt;
```

Comment:
@mximp this should also be `final`. IDE must hint you about this.

Reply:
@mximp in general, non-`final` variables is a code smell. Try to avoid them at all cost.